src/code/z_camera.c

```diff
@@ -1,3069 +1,6777 @@
+            camera->inputDir.x = adjSph.pitch;
+            camera->inputDir.y = adjSph.yaw;
+            camera->inputDir.z = 0;
+        }
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_camera/Camera_Fixed2.s")
+    camera->fov = Camera_LERPCeilF(anim->unk_1C, camera->fov, camera->fovUpdateRate, 1.0f);
+    camera->roll = Camera_LERPCeilS(0, camera->roll, 0.5f, 0xA);
+    camera->atLERPStepScale = Camera_ClampLERPScale(camera, norm2->unk_18);
+    return 1;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_camera/Camera_Fixed3.s")
+// riding epona
+s32 Camera_Normal3(Camera* camera) {
+    Vec3f* eye = &camera->eye;
+    Vec3f* at = &camera->at;
+    Vec3f* eyeNext = &camera->eyeNext;
+    f32 sp98;
+    f32 sp94;
+    f32 sp90;
+    f32 sp8C;
+    VecSph sp84;
+    VecSph sp7C;
+    VecSph sp74;
+    PosRot* playerPosRot = &camera->playerPosRot;
+    f32 temp_f0;
+    f32 temp_f6;
+    s16 phi_a0;
+    s16 t2;
+    Normal3* norm3 = (Normal3*)camera->paramData;
+    Normal3Anim* anim = &norm3->anim;
+    f32 playerHeight;
+
+    playerHeight = Player_GetHeight(camera->player);
+    if (RELOAD_PARAMS) {
+        CameraModeValue* values = sCameraSettings[camera->setting].cameraModes[camera->mode].values;
+        norm3->yOffset = NEXTSETTING * PCT(playerHeight);
+        norm3->distMin = NEXTSETTING * PCT(playerHeight);
+        norm3->distMax = NEXTSETTING * PCT(playerHeight);
+        norm3->pitchTarget = DEGF_TO_BINANG(NEXTSETTING);
+        norm3->yawUpdateSpeed = NEXTSETTING;
+        norm3->unk_10 = NEXTSETTING;
+        norm3->fovTarget = NEXTSETTING;
+        norm3->maxAtLERPScale = NEXTPCT;
+        norm3->interfaceFlags = NEXTSETTING;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_camera/Camera_Fixed4.s")
+    if (R_RELOAD_CAM_PARAMS) {
+        Camera_CopyPREGToModeValues(camera);
+    }
 
-s32 Camera_Fixed0(Camera* camera) {
-    return func_80047394(camera);
-}
+    OLib_Vec3fDiffToVecSphGeo(&sp7C, at, eye);
+    OLib_Vec3fDiffToVecSphGeo(&sp74, at, eyeNext);
 
-s32 Camera_Subj1(Camera* camera) {
-    return func_80047394(camera);
-}
+    sUpdateCameraDirection = true;
+    sCameraInterfaceFlags = norm3->interfaceFlags;
+    switch (camera->animState) {
+        case 0:
+        case 0xA:
+        case 0x14:
+        case 0x19:
+            anim->swing.atEyePoly = NULL;
+            anim->curPitch = 0;
+            anim->unk_1C = 0.0f;
+            anim->unk_20 = camera->playerGroundY;
+            anim->swing.unk_16 = anim->swing.unk_14 = anim->swing.unk_18 = 0;
+            anim->swing.swingUpdateRate = norm3->yawUpdateSpeed;
+            anim->yawUpdAmt = BINANG_SUB(BINANG_ROT180(playerPosRot->rot.y), sp7C.yaw) * (1.0f / OREG(23));
+            anim->distTimer = 10;
+            anim->yawTimer = OREG(23);
+            camera->animState = 1;
+            anim->swing.swingUpdateRateTimer = 0;
+    }
 
-s32 Camera_Subj2(Camera* camera) {
-    return func_80047394(camera);
-}
+    DECR(anim->distTimer);
```

Comment:
We generally avoid DECR, if possible ofc
```suggestion
    if (anim->distTimer != 0) {
        anim->distTimer--;
    }
```